testutils/src/main/scala/za/co/absa/enceladus/testutils/infoFileComparison/AtumModelUtils.scala

```diff
@@ -0,0 +1,192 @@
+    */
+  implicit class ControlMeasureOps (controlMeasure: ControlMeasure){
+    /**
+      * Compare this ControLMeasure with the one passed in
+      * @param secondControlMeasure Second control measure
+      * @return Returns a list of model differences
+      */
+    def compareWith(secondControlMeasure: ControlMeasure): List[ModelDifference[_]] = {
+      val metadataDifferences = controlMeasure.metadata.compareWith(secondControlMeasure.metadata, "metadata")
+      val checkpointsDifferences = controlMeasure.checkpoints.zipWithIndex.foldLeft(List[ModelDifference[_]]()) {
+        case (agg, (value, index)) =>
+          value.compareWith(secondControlMeasure.checkpoints(index), s"checkpoints[$index]") ::: agg
+      }
+      metadataDifferences ::: checkpointsDifferences
+    }
+  }
+
+  /**
+    * ControlMeasureMetadata's class extension adding compareWith
+    * @param metadata ControlMeasureMetadata instance
+    */
+  implicit class ControlMeasureMetadataOps (metadata: ControlMeasureMetadata){
+    /**
+      * Compare this ControlMeasureMetadata with the one passed in
+      * @param secondMetadata Second control measure metadata
+      * @param curPath Path to the ControlMeasureMetadata through the model
+      * @return Returns a list of model differences
+      */
+    def compareWith(secondMetadata: ControlMeasureMetadata, curPath: String): List[ModelDifference[_]] = {
+      val diffs = simpleCompare(metadata.sourceApplication,
+        secondMetadata.sourceApplication, s"$curPath.sourceApplication") ::
+      simpleCompare(metadata.country, secondMetadata.country, s"$curPath.country") ::
+      simpleCompare(metadata.historyType, secondMetadata.historyType, s"$curPath.historyType") ::
+      simpleCompare(metadata.dataFilename, secondMetadata.dataFilename, s"$curPath.dataFilename") ::
+      simpleCompare(metadata.sourceType, secondMetadata.sourceType, s"$curPath.sourceType") ::
+      simpleCompare(metadata.version, secondMetadata.version, s"$curPath.version") ::
+      simpleCompare(metadata.informationDate, secondMetadata.informationDate, s"$curPath.informationDate") ::
+        Nil
+
+      val additionalInfoDiff = additionalInfoComparison(metadata.additionalInfo,
+        secondMetadata.additionalInfo,
+        s"$curPath.additionalInfo")
+
+      (diffs flatten) ::: additionalInfoDiff
+    }
+
+    /**
+      * Compare additional information from ControlMeasureMetadata
+      * @param was     Value it had in reference
+      * @param is Value it has now
+      * @param curPath Path to the AdditionalInfo through the model
+      * @return Returns a list of model differences
+      */
+    private def additionalInfoComparison(was: Map[String, String],
+                                         is: Map[String,String],
+                                         curPath: String): List[ModelDifference[_]] = {
+      was.flatMap {
+        case (wasKey, wasValue) if wasKey == "conform_enceladus_version" =>
```

Comment:
I have a feeling this might be abusing pattern matching in a very seductive way. The guard clauses can easily be turned into a regular `if-else` thing, but I think this is much more readable. I like it quite a bit and would keep it as is, but I'm curious to get @wajda 's opinion (bring it up in CQC?)

Reply:
I also like it as is